firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -17,18 +17,37 @@
 import android.os.Bundle;
 import androidx.annotation.NonNull;
 import com.google.firebase.analytics.connector.AnalyticsConnector;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import org.json.JSONObject;
 
 public class Personalization {
   public static final String ANALYTICS_ORIGIN_PERSONALIZATION = "fp";
-  public static final String ANALYTICS_PULL_EVENT = "_fpc";
-  public static final String ARM_KEY = "_fpid";
-  public static final String ARM_VALUE = "_fpct";
-  static final String PERSONALIZATION_ID = "personalizationId";
+
+  // Constants with PARAM suffix are how the corresponding ones without it are identified on
+  // Google Analytics.
+  public static final String EXTERNAL_EVENT = "personalization_assignment";
+  public static final String EXTERNAL_RC_PARAMETER_PARAM = "arm_key";
+  public static final String EXTERNAL_ARM_VALUE_PARAM = "arm_value";
+  public static final String PERSONALIZATION_ID = "personalizationId";
+  public static final String EXTERNAL_PERSONALIZATION_ID_PARAM = "personalization_id";
+  public static final String ARM_INDEX = "armIndex";
+  public static final String EXTERNAL_ARM_INDEX_PARAM = "arm_index";
+  public static final String GROUP = "group";
+  public static final String EXTERNAL_GROUP_PARAM = "group";
+
+  public static final String INTERNAL_EVENT = "_fpc";
+  public static final String CHOICE_ID = "choiceId";
+  public static final String INTERNAL_CHOICE_ID_PARAM = "_fpid";
 
   /** The app's Firebase Analytics client. */
   private final AnalyticsConnector analyticsConnector;
 
+  /** A map of Remote Config parameter key to choice ID. */
```

Comment:
Related to removing the comment below about reducing logging - I'd just note that here as something like "Remote Config parameter key and Choice ID pairs that have already been logged to Analytics."